code/processes/dqe.q

```diff
@@ -7,11 +7,46 @@ init:{
 
 gethandles:{exec procname,proctype,w from .servers.SERVERS where (procname in x) | (proctype in x)};
 
-tableexists:{[tabs]                                                                                             /- function to check for table, param is table name as a symbol
+tableexists:{[tab]                                                                                              /- function to check for table, param is table name as a symbol
   result:();
-  $[1=all a:tabs in tables[];result:(1b;"all tables exist");result:(0b;raze ("," sv string tabs where not a)," missing from process")]
+  .lg.o[`tabexist;"checking if ", (string tab), " table exists"];
+  $[1=a:tab in tables[];result:(1b;((string tab)," table exists"));result:(0b;(string tab)," missing from process")]
   };
 
+fillprocname:{[h;rs]                                                                                            /- fill procname for results table
+  $[0=first where rs in ' h`procname`proctype;
+    enlist rs,'rs;
+    1=first where rs in ' h`procname`proctype;
+    rs,'exec procname from flip h where proctype=rs;enlist rs,'`]
+  }
+
+initstatusupd:{[id;funct;vars;rs]                                                                               /- set initial values in results table
+  .lg.o[`results;"setting up initial record(s) for id ",(string id)];
+  `.dqe.results insert (id;funct;`$"," sv string (),vars;rs[0];rs[1];.z.p;0Np;`;"";`started);
+  }
+
+failunconnected:{[idnum;proc]
+  .lg.o[`results;raze "run check id ",(string idnum)," update in results table with a fail as can't connect"];
+  update chkstatus:`failed,descp:enlist "error:can't connect to process" from `.dqe.results where id=idnum, procs=proc;
+  }
+
+failerror:{[idnum;proc;error]
+ .lg.o[`results;raze "run check id ",(string idnum)," update in results table with a fail, with ",(string error)];
+ update chkstatus:`failed,descp:enlist error from `.dqe.results where id=idnum, procschk=proc;
+ }
+
+postback:{[idnum;proc;result]
+  $["e"=first result;
+  .dqe.failerror[idnum;proc;result];
+  update endtime:.z.p,output:`$ string first result,descp:enlist last result,chkstatus:`complete from `.dqe.results where id=idnum,procschk=proc];
+  show result
+  }
+
+getresult:{[funct;vars;id;proc;hand]
+  .lg.o[`results;raze"send function over to prcess: ",(string proc)];
```

Comment:
.lg.o structure what should the first parameter be used for?  Is it for a meaningful name or for the function name?

Reply:
It is the name that will appear in between the |, just before the error message.
I believe it is used as a meaningful name so that you can figure out where it run from.
```
2019.11.29D14:58:10.840435000|peter|dqe|dqe1|INF|results|send function over to prcess: hdb2
```
Not completely sure though :)

Reply:
```
q).api.f".lg.[oe]"
name | vartype  namespace public descrip               params                                         return
-----| -----------------------------------------------------------------------------------------------------
.lg.o| function .lg       1      "Log to standard out" "[symbol: id of log message; string: message]" "null"
.lg.e| function .lg       1      "Log to standard err" "[symbol: id of log message; string: message]" "null"
```

api docs describe it as "id of log message", I think it's generally used to indicate where in the code log message is happening (e.g. which function)